### PR TITLE
regclient blackbox tests and regclient installation in Makefile

### PR DESCRIPTION
**Pull request by aokirisaki**
**What type of PR is this?**

**Which issue does this PR fix**:
#586 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by peusebiu — DISMISSED** [review]

File: test/blackbox/pushpull.bats [hunk 1]
```diff
@@ -69,23 +69,73 @@ function teardown_file() {
...
+    [ "$status" -eq 0 ]
 }
 
 @test "pull helm chart" {
-    helm pull oci://localhost:8080/zot-chart/zot --version 0.1.0
+    run helm pull oci://localhost:8080/zot-chart/zot --version 0.1.0
+    [ "$status" -eq 0 ]
+}
+
+@test "pull image with regclient" {
```
Comment (andaaron) [inline test/blackbox/pushpull.bats:98]: "pull image"

File: test/blackbox/pushpull.bats [hunk 2]
```diff
@@ -69,23 +69,73 @@ function teardown_file() {
 @test "push oras artifact" {
     echo "{\"name\":\"foo\",\"value\":\"bar\"}" > config.json
     echo "hello world" > artifact.txt
-    oras push --plain-http 127.0.0.1:8080/hello-artifact:v2 \
+    run oras push --plain-http 127.0.0.1:8080/hello-artifact:v2 \
         --manifest-config config.json:application/vnd.acme.rocket.config.v1+json artifact.txt:text/plain -d -v
-	rm -f artifact.txt
+	[ "$status" -eq 0 ]
```
Comment (andaaron) [inline test/blackbox/pushpull.bats:74]: Please fix indentation

File: test/blackbox/pushpull.bats (hunk 1 above)
Comment (peusebiu) [reply test/blackbox/pushpull.bats:98]: also add a pull image test :D

**Review by andaaron — DISMISSED** [review]
LGTM

**Review by andaaron — DISMISSED** [review]

**Review by peusebiu — APPROVED** [review]
LGTM

**Review by andaaron — APPROVED** [review]

**Comment by andaaron** [comment]
Merged, the changes do not impact go code, so the coverage report had an error.

**Pull request merged**